src/System.Private.Xml.Linq/src/System/Xml/Linq/XHashtable.cs

```diff
@@ -109,17 +109,7 @@ public TValue Add(TValue value)
                 lock (this)
                 {
                     XHashtableState newState = _state.Resize();
-
-                    // Use memory barrier to ensure that the resized XHashtableState object is fully constructed before it is assigned
-#if !SILVERLIGHT 
-                    Thread.MemoryBarrier();
-#else // SILVERLIGHT
-                    // The MemoryBarrier method usage is probably incorrect and should be removed.
-
-                    // Replacing with Interlocked.CompareExchange for now (with no effect)
-                    //   which will do a very similar thing to MemoryBarrier (it's just slower)
```

Comment:
This should be just changed to use `Thread.MemoryBarrier`. The hard to understand Silverlight hack is not necessary. `Thread.MemoryBarrier` available everywhere now, so we can just use it.

Reply:
addressed